### PR TITLE
Try a fix for the spec failures, with an API change side-effect

### PR DESCRIPTION
**Pull request by kgrz**
TL;DR

Should we make static files that get served by `send_file` be restricted to `public` directory post 2.0?

Changes:

**Set the Rack::File's root to the app file's root**

`Rack::File`'s API doesn't accept a `nil` for root path. Earlier, we
were assigning the path of the file manually in Sinatra by the following
steps:

```ruby
file = Rack::File.new(nil)
file.path = path
file.serving(env)
```

But the API of `Rack::File#serving` has been changed to take two
arguments: `request` and `path` of the file. `Rack::File` would then
interpret the file path based on the root path supplied to it's `#new`
method, and the `path` value passed into the `#serving` method.

For these reasons, we decided to use `Rack::File.call` method, but this
causes issues with how the root path is set as `Rack::File` since it
appends the route fragment to the root path to fetch the file. This has
two side-effects:

1. The static file that's being served should be present in the public
directory
2. The static file path under `public` directory should match the route
fragment, including the extension.

For example:

```ruby
get '/file.txt' do
  send_file 'file.txt'
end
```

will fetch the file at `<app root>/public/file.txt`, whereas:

```ruby
get '/file' do
  send_file 'file.txt'
end
```

will try to fetch the file at `<app root>/public/file` and will fail
since there won't be a `file.txt` at that location.

This PR also fixes the failing specs that are caused by this appending
of `public` directory effect by simply setting the public directory to
app file root by using `settings.app_file` method.

**Revert 855208a2936b27da68588ba547b3fb70c3e48268**

This causes conflicts with the way the instance variable `@file` in the
test file works with the scope of `mock_app`. Without assigning the
`@file` instance variable to a temporary variable—in this case `path`—we
won't be able to access the instance var inside the `mock_app`'s `get`
context.

### CODE REVIEW

**Comment by kgrz** [comment]
The one spec that's failing is related to `Rack::File`s `ALLOWED_VERBS` constant that doesn't allow `POST` request. https://github.com/rack/rack/blob/master/lib/rack/file.rb#L16

If I add the `'POST'` string to that array, the failing spec passes. I'm not sure how to go about this. Subclass like ShowExceptions?
